nevergrad/benchmark/experiments.py

```diff
@@ -1257,5 +1299,61 @@ def bragg_structure(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
         for algo in default_optims if default_optims is not None else optims:
             yield Experiment(func, algo, int(budget), num_workers=1, seed=xpseed)
         for f in [func, func_nostruct, func_mix]:
-            for algo2 in recombinable:
-                yield Experiment(f, algo2, int(budget), num_workers=1, seed=xpseed)
+            for algo in recombinable:
+                yield Experiment(f, algo, int(budget), num_workers=1, seed=xpseed)
+
+
+@registry.register
+def adversarial_attack(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
+    class Normalize(nn.Module):
+        def __init__(self, mean, std):
+            super().__init__()
+            self.mean = torch.Tensor(mean)
+            self.std = torch.Tensor(std)
+
+        def forward(self, x):
+            return (x - self.mean.type_as(x)[None, :, None, None]) / self.std.type_as(x)[None, :, None, None]
+
+    class Resnet50(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.norm = Normalize(mean=[0.485, 0.456, 0.406],
+                                  std=[0.229, 0.224, 0.225])
+            self.model = resnet50(pretrained=True)
+
+        def forward(self, x):
+            return self.model(self.norm(x))
+
+    image_size = 224
+    data_folder = "/datasets01_101/imagenet_full_size/061417/val"
+    if not os.path.exists(data_folder):
+        print("this path does not exist")
+        return
```

Comment:
I see two solutions.
Solution 1: Use a trivial data loader with one single image. This is the cleanest solution, as it has a better code coverage.
Solution 2: Easier. Maybe define an exception, of the form ExperimentSkippedOnCI. Instead of "return", do "raise ExperimentSkippedOnCI", and in the test you do <<try: .... except ExperimentSkippedOnCI: pass>>.

Reply:
solved